app/src/main/java/nz/org/cacophony/cacophonometer/TestRecordFragment.java

```diff
@@ -0,0 +1,195 @@
+
+        } catch (Exception ex) {
+            Log.e(TAG, ex.getLocalizedMessage());
+        }
+    }
+
+    private final BroadcastReceiver onNotice = new BroadcastReceiver() {
+
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (getView() == null) {
+                return;
+            }
+
+            try {
+
+                String jsonStringMessage = intent.getStringExtra("jsonStringMessage");
+
+                if (jsonStringMessage != null) {
+
+                    JSONObject joMessage = new JSONObject(jsonStringMessage);
+                    String messageType = joMessage.getString("messageType");
+                    String messageToDisplay = joMessage.getString("messageToDisplay");
+
+                    if (messageType.equalsIgnoreCase("RECORDING_DISABLED")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("ALREADY_RECORDING")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("NO_PERMISSION_TO_RECORD")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("UPLOADING_RECORDINGS")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("UPLOADING_FAILED")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("UPLOADING_FINISHED")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("SUCCESSFULLY_UPLOADED_RECORDINGS")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("GETTING_READY_TO_RECORD")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("FAILED_RECORDINGS_NOT_UPLOADED")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("RECORD_AND_UPLOAD_FAILED")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("UPLOADING_FAILED_NOT_REGISTERED")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("RECORDING_STARTED")) {
+                        tvMessages.setText(messageToDisplay);
+                    } else if (messageType.equalsIgnoreCase("RECORDING_FINISHED")) {
+                        tvMessages.setText(messageToDisplay);
+                        //getView().findViewById(R.id.btnRecordNow).setEnabled(true);
+                        btnRecordNow.setEnabled(true);
+                        btnRecordNow.setVisibility(View.VISIBLE);
+                    }
```

Comment:
if messageType is made uppercase first then you could use a switch statement here instead. That would probably be a little less verbose.

Reply:
The current version of Java that Android uses does not allow switch with Strings.  I probably should have used enumerations... but won't change now.
https://stackoverflow.com/questions/14367629/android-coding-with-switch-string

Reply:
All good. This shows my relative inexperience with Java :)